src/test/scala/HeadTest.scala

```diff
@@ -16,4 +16,36 @@ class HeadTest extends FunSuite with Matchers with PropertyChecks {
       }
     }
   }
+
+  test("Head should be first element of input"){
+    forAll{ input: List[Int] =>
+      whenever (input.nonEmpty){
+        input(0) == Head(input) should be (true)
+      }
+    }
+  }
+
 }
+
+class FilterTest extends FunSuite with Matchers with PropertyChecks {
+
+  test("Filter should return a list smaller or equal in size to the input") {
+    forAll { (input: List[Int], pred: Boolean) => {
+      Filter(input, { X: Int => pred }).size <= input.size} should be(true)
```

Comment:
this is good. you've got the totally permissive 'accept anything' function of {_ => true} and the 'remove everything' function of {_ => false}. So you're checking both extremes with arbitrary lists of Int's. excellent.